### PR TITLE
Add delete record function

### PR DESCRIPTION
**Pull request by sakshambassi**
This PR contains the addition of `Delete inventory record` functionality. 

1. Functionality

Add `delete_inventory_record` to delete an existing record.

2. Tested the following changes using unit tests : 

- `test_delete_inventory_record_success`
- `test_delete_inventory_record_exception` 

Code coverage: 
`service/models 94%`
`service/routes 100%`

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -86,6 +87,26 @@ def list_inventory_records():
...
+def delete_inventory_record(product_id):
+    """Deletes inventory record
+
+    @param: product_id is the id of the record that is to be deleted
+    """
+    app.logger.info("Request to delete inventory record with id: %s", product_id)
+    inventory = Inventory()
+    inventory.deserialize(request.get_json())
+    inventory_record = inventory.find((inventory.product_id, inventory.condition))
+    app.logger.info("inventory_recordinventory_record", inventory_record)
```
Comment (richa15997) [inline service/routes.py:100]: maybe we can rename the info to -> "For Deleting checking if Inventory Record Exists" or something more informative

Comment (sakshambassi) [reply service/routes.py:100]: Thanks for your comment, will update!

Comment (sakshambassi) [reply service/routes.py]: Added to new commit, resolved!

**Review by richa15997 — APPROVED** [review]
Awesome!

**Review by rishit13 — DISMISSED** [review]

**Review by richa15997 — APPROVED** [review]

**Pull request merged**